### PR TITLE
Support reserve dir delete if pooling is disabled

### PR DESCRIPTION
**Pull request by zzmao**
Delete reserved dir if DiskSpaceAllocator pooling is disabled.

### CODE REVIEW

File: ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java [hunk 1]
```diff
@@ -101,6 +103,12 @@
         prepareDirectory(swapReserveDir);
         inventoryExistingReserveFiles();
         poolState = PoolState.INVENTORIED;
+      } else {
+        // Clean up reserve directory if previously created.
+        if (reserveDir != null) {
+          deleteDirectory(reserveDir);
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java:109]: you can use `deleteFileOrDirectory()` in `Utils.java`

File: ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java [hunk 2]
```diff
@@ -101,6 +103,12 @@
         prepareDirectory(swapReserveDir);
         inventoryExistingReserveFiles();
         poolState = PoolState.INVENTORIED;
+      } else {
+        // Clean up reserve directory if previously created.
+        if (reserveDir != null) {
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java:108]: Can we check `if(reserveDir != null && reserveDir.exists())` ?

File: ambry-store/src/test/java/com/github/ambry/store/DiskSpaceAllocatorTest.java [hunk 3]
```diff
@@ -436,6 +436,26 @@ public void unpooledAllocatorTest() throws Exception {
...
+    assertTrue("Reserve directory should exist", reserveFileDir.exists());
+
+    // Then construct unpooled allocator, reserved directories should be deleted.
+    alloc = constructUnpooledAllocator();
+    verifyPoolState(null);
+    File f1 = allocateAndVerify(storeId0, "file1", 20, false);
+    verifyPoolState(null);
+    freeAndVerify(storeId0, f1, 20, false);
+    verifyPoolState(null);
+    assertFalse("Reserve directory should not exist", reserveFileDir.exists());
```
Comment (cgtz) [inline ambry-store/src/test/java/com/github/ambry/store/DiskSpaceAllocatorTest.java:456]: Could you test recreating a pooled allocator after the unpooled allocator was used?

**Review by cgtz — COMMENTED** [review]
LGTM after Yingyi's comments are addressed.

File: ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java (hunk 1 above)
Comment (zzmao) [reply ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java]: changed to our `deleteFileOrDirectory `

File: ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java (hunk 2 above)
Comment (zzmao) [reply ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java:106]: `deleteFileOrDirectory ` checks `exists ` first. So we don't need an extra check here.

File: ambry-store/src/test/java/com/github/ambry/store/DiskSpaceAllocatorTest.java (hunk 3 above)
Comment (zzmao) [reply ambry-store/src/test/java/com/github/ambry/store/DiskSpaceAllocatorTest.java:457]: test added.

**Review by jsjtzyy — APPROVED** [review]

**Comment by zzmao** [comment]
@cgtz  can we merge this ? (though we don't need this probably)

**Pull request merged**